instrumentation/apache-pulsar/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/telemetry/ProducerAttributeGetter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pulsar.telemetry;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessagingAttributesGetter;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import javax.annotation.Nullable;
+import org.apache.pulsar.client.api.Message;
+
+class ProducerAttributeGetter implements MessagingAttributesGetter<Message<?>, Attributes> {
```

Comment:
can this be a single implementation of MessagingAttributesGetter and remove the other 2?
```suggestion
class PulsarMessagingAttributesGetter implements MessagingAttributesGetter<Message<?>, Attributes> {
```